### PR TITLE
Expanded input event examples

### PR DESCRIPTION
**Pull request by EgoOrb**
This PR adds some detail to the input event examples to make them a bit more useful at a glance.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Cool! I didn't know about ElementState; that's a nice abstraction.

File: examples/input/keyboard_input_events.rs [hunk 1]
```diff
@@ -10,6 +10,18 @@ fn main() {
 /// This system prints out all keyboard events as they come in
 fn print_keyboard_event_system(mut keyboard_input_events: EventReader<KeyboardInput>) {
     for event in keyboard_input_events.iter() {
-        info!("{:?}", event);
+        let scan_code: u32 = event.scan_code;
+        let key_code: Option<KeyCode> = event.key_code;
+        let state: ElementState = event.state;
+        info!("Scan code: {}", scan_code);
+        info!("Key code: {:?}", key_code);
+        match state {
```
Comment (mockersf) [inline examples/input/keyboard_input_events.rs:18]:
```suggestion
        info!("Scan code: {}", event.scan_code);
        info!("Key code: {:?}", event.key_code);
        match event.state {
```

Comment (EgoOrb) [reply examples/input/keyboard_input_events.rs:18]: I chose to declare those variables to make it obvious what type each field is. It's a little more verbose, but I think it adds a bit of clarity.

Comment (alice-i-cecile) [reply examples/input/keyboard_input_events.rs:18]: I think I slightly prefer Francois's approach, but it's just a matter of style. I do like how your code lays out the types explicitly though.

**Comment by alice-i-cecile** [comment]
I'm going to toss ready-for-cart on this; I think it's just one question of style as a tie-breaker.

File: examples/input/keyboard_input_events.rs (hunk 1 above)
Comment (cart) [reply examples/input/keyboard_input_events.rs:18]: I also prefer the "inline style" recommended by @mockersf. Thats the style I would encourage developers to use, and I like examples to encourage "best practices" where possible. Minimalism / keeping examples small is also a priority.

Comment (EgoOrb) [reply examples/input/keyboard_input_events.rs]: Done.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by cart** [comment]
Coming back to this: I think this is actually a net loss for a number of reasons:

1. If we add new fields to input events they won't be automatically picked up in examples
2. This is largely just rearranging data that is already there / rephrasing variable names. It doesn't feel like a major increase in legibility. Ex:
    * Before:
       image
    * After:
       image
3. Breaking events up into multiple log lines makes it harder to tell what values correspond to each event
4. It adds more code to tests, when our goal is to make them as minimal as possible.

I feel strongly enough about this that I'm closing the issue, but feel free to continue the discussion. I'll reopen if anyone pushes back with compelling counter arguments.

**Pull request closed without merging**